packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -124,6 +132,7 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
         // These are set in ascending order of their UpdateTypes.
         _setBatcherHash(_batcherHash);
         _setGasConfig({ _overhead: _overhead, _scalar: _scalar });
+        _setGasConfigEcotone({ _basefeeScalar: uint32(_scalar), _blobBasefeeScalar: 0 });
```

Comment:
I noted the type casting in the specs https://github.com/ethereum-optimism/optimism/pull/8829

This is simple enough behavior, if the scalar is larger than `type(uint32).max` then use `type(uint32).max`